File: unix/boot/spp/rpp/rppfor/outnum.f
      subroutine outnum (n)
      integer n
      integer chars (20)
      integer i, m
      m = iabs (n)
      i = 0
23000 continue
      i = i + 1
      chars (i) = mod (m, 10) + 48
      m = m / 10
23001 if (.not.(m .eq. 0 .or. i .ge. 20))goto 23000
23002 continue
      if (.not.(n .lt. 0))goto 23003
      call outch (45)
23003 continue
23005 if (.not.(i .gt. 0))goto 23007
      call outch (chars (i))
23006 i = i - 1
      goto 23005
23007 continue
      return
      end

File: unix/boot/spp/rpp/rpprat/outnum.r
#-h-  outnum			  381  local   12/01/80  15:54:32
# outnum - output decimal number
   include  defs

   subroutine outnum (n)
   integer n

   character chars (MAXCHARS)

   integer i, m

   m = iabs (n)
   i = 0
   repeat {
      i = i + 1
      chars (i) = mod (m, 10) + DIG0
      m = m / 10
      } until (m == 0 | i >= MAXCHARS)
   if (n < 0)
      call outch (MINUS)
   for ( ; i > 0; i = i - 1)
      call outch (chars (i))
   return
   end
